code/lib/cli/src/generators/baseGenerator.ts

```diff
@@ -45,15 +45,32 @@ const getBuilderDetails = (builder: string) => {
 
 const getExternalFramework = (framework: string) =>
   externalFrameworks.find(
-    (exFramework) => exFramework.name === framework || exFramework.packageName === framework
+    (exFramework) =>
+      framework !== undefined &&
```

Comment:
> The logic around external frameworks is getting a bit convoluted, but I think that's okay since all of this should only be temporary until we move external frameworks out of the CLI altogether.

I agree. It needs some refactoring in the future once you have standardized the new way for handling frameworks.

> why would framework be undefined here, parameter typings suggests it can't be?

I think it's because the' undefined' type is considered in `ts` a subtype of all other types, including string.
For this reason, ts doesn't complain at line `92` when receives `framework?` from `getFrameworkDetails` fn. So I needed to add this possible scenario when `framework` is `undefined`.